mne/coreg.py

```diff
@@ -1245,3 +1248,632 @@ def _scale_xfm(subject_to, xfm_fname, mri_name, subject_from, scale,
+        high_res_path = _find_head_bem(subject, subjects_dir, high_res=True)
+        low_res_path = _find_head_bem(subject, subjects_dir, high_res=False)
+        if high_res_path is None and low_res_path is None:
+            raise RuntimeError("No standard head model was "
+                               f"found for subject {subject}")
+        if high_res_path is not None:
+            self.bem_high_res = self._read_surface(high_res_path)
+        else:
+            self.bem_high_res = self._read_surface(low_res_path)
+        if low_res_path is None:
+            # This should be very rare!
+            warn('No low-resolution head found, decimating high resolution '
+                 'mesh (%d vertices): %s' % (len(self.bem_high_res.surf.rr),
+                                             high_res_path,))
+            # Create one from the high res one, which we know we have
+            rr, tris = decimate_surface(self.bem_high_res.surf.rr,
+                                        self.bem_high_res.surf.tris,
+                                        n_triangles=5120)
+            # directly set the attributes of bem_low_res
+            self.bem_low_res = complete_surface_info(dict(rr=rr, tris=tris),
+                                                     copy=False, verbose=False)
+        else:
+            self.bem_low_res = self._read_surface(low_res_path)
+
+        # Set MNI points
+        try:
+            fids = get_mni_fiducials(subject, subjects_dir)
+        except Exception:  # some problem, leave at origin
+            self.mni_points = None
+        else:
+            self.mni_points = np.array([f['r'] for f in fids], float)
+
+        # find fiducials file
+        fid_files = _find_fiducials_files(subject, subjects_dir)
+        if len(fid_files) > 0:
+            fid_filename = fid_files[0].format(subjects_dir=subjects_dir,
+                                               subject=subject)
+        else:
+            fid_filename = None
+        if fid_filename is None or not op.exists(fid_filename):
+            self.fid_points = self.mni_points
+        else:
+            self.fid_points = _fiducial_coords(*read_fiducials(fid_filename))
+
+        # does not seem to happen by itself ... so hard code it:
+        self._reset_fiducials()
+
+    def _read_surface(self, filename):
+        bem = dict()
+        if filename is not None and op.exists(filename):
+            if filename.endswith('.fif'):
+                bem = read_bem_surfaces(filename, verbose=False)[0]
+            else:
+                try:
+                    bem = read_surface(filename, return_dict=True)[2]
+                    bem['rr'] *= 1e-3
+                    complete_surface_info(bem, copy=False)
+                except Exception:
+                    raise ValueError(
+                        "Error loading surface from %s (see "
+                        "Terminal for details)." % filename)
+        return bem
```

Comment:
This function does not use self so it can be made a private function outside of the class